### PR TITLE
upgrading dependencies and fixing linting errors

### PR DESCRIPTION
**Pull request by dlockhart**
Another bit of quick cleanup to get things into shape.

### CODE REVIEW

File: logging.js [hunk 1]
```diff
@@ -83,15 +83,13 @@ export class ServerLogger {
...
 		this._log = { appId: String(appId) };
 	}
 
-	withLocation() {
-		if (window.location && window.location.href) {
-			this._log.location = String(window.location.href);
-		}
-		return this;
+	build() {
+		return Object.freeze(this._log);
```
Comment (dlockhart) [inline logging.js:92]: These are all just alphabetizing method names, which is required as part of the new linting reules.

File: test/logging.test.js [hunk 2]
```diff
@@ -128,15 +127,13 @@ describe('logging', () => {
 							expect(client._uniqueLogs).to.be.empty;
 							done();
 						}
-					};
-					const client = new LoggingClient('my-app-id', mockLogger);
+					});
```
Comment (dlockhart) [inline test/logging.test.js:130]: The `mockLogger` was using `client` before it was defined. So moving it down into the same line fixes that.

**Review by svanherk — APPROVED** [review]
🎉

**Pull request merged**